uaclient/tests/test_apt.py

```diff
@@ -152,7 +153,34 @@ def test_errors_on_process_execution_errors(
         apt_helper_call = mock.call(
             ['/usr/lib/apt/apt-helper', 'download-file',
              'http://user:pwd@fakerepo/ubuntu/pool/',
-             '/tmp/uaclient-apt-test'])
+             '/tmp/uaclient-apt-test'], timeout=20)
+        assert [apt_helper_call] == m_subp.call_args_list
+        assert [mock.call('/tmp/uaclient-apt-test')] == m_unlink.call_args_list
+
+    @mock.patch('uaclient.apt.os.unlink', return_value=True)
+    @mock.patch('uaclient.util.subp')
+    @mock.patch('uaclient.apt.os.path.exists', return_value=True)
+    def test_errors_on_apt_helper_process_timeout(
+            self, m_exists, m_subp, m_unlink):
+        """Raise the appropriate user facing error from apt-helper timeout."""
+
+        # Failure apt-helper response
+        m_subp.side_effect = subprocess.TimeoutExpired(
+            'something timed out', timeout=1000000)
+        with pytest.raises(exceptions.UserFacingError) as excinfo:
+            assert_valid_apt_credentials(
+                repo_url='http://fakerepo', username='user', password='pwd')
+        error_msg = (
+            'Cannot validate credentials for APT repo: fakerepo. Timeout'
```

Comment:
If you're concerned about a long repo URL (which is definitely a valid concern!), we could consider making this something like:

ERROR: Cannot validate credentials for APT repo. Timeout after 20 seconds trying to reach https://esm.ubuntu.com/this/is/a/long/url/but/its/at/the/end/so/its/not/so/bad.

Reply:
+1